apps/functions/applications-migration/folder-migration/src/folder-migration.js

```diff
@@ -0,0 +1,59 @@
+import { SynapseDB } from '../../common/synapse-db.js';
+import { makePostRequest } from '../../common/back-office-api-client.js';
+import { removeValues } from '../../common/utils.js';
+
+const query = 'SELECT * FROM [odw_curated_db].[dbo].[legacy_folder_data] WHERE caseReference = ?';
+
+/**
+ * Handle an HTTP trigger/request to run the migration
+ *
+ * @param {import('@azure/functions').Logger} logger
+ * @param {string[]} caseReferences
+ */
+export const migrateFolder = async (logger, caseReferences) => {
+	for (const caseReference of caseReferences) {
+		try {
+			logger.info(`reading Folders with caseReference ${caseReference}`);
+
+			const [folderRows, count] = await SynapseDB.query(query, {
+				replacements: [caseReference]
+			});
+
+			const folderEntities = folderRows.map((row) => {
+				const isRootFolder = row.parentFolderId === row.caseNodeId;
+
+				const entity = {
+					id: row.id,
+					caseReference: row.caseReference,
+					displayNameEnglish: row.displayNameEnglish,
+					parentFolderId: isRootFolder ? null : row.parentFolderId,
+					caseStage: mapCaseStage(row.caseStage)
+				};
+
+				removeValues(entity, [null, 'None']);
+
+				return entity;
+			});
+
+			logger.info(`found ${count} Folders: ${JSON.stringify(folderEntities.map((u) => u.id))}`);
+
+			if (folderEntities.length > 0) {
+				await makePostRequest(logger, '/migration/folder', folderEntities);
+			}
+		} catch (e) {
+			logger.error(`Failed to migrate Folder for case ${caseReference}`, e?.response?.body, e);
+			throw e;
+		}
+	}
+};
+
+const mapCaseStage = (caseStage) =>
+	({
+		'Pre-application': 'pre-application',
+		Acceptance: 'acceptance',
+		'Pre-examination': 'pre-examination',
+		Examination: 'examination',
+		Recommendation: 'recommendation',
+		Decision: 'decision',
+		'Post decision': 'post_decision'
+	}[caseStage] || caseStage);
```

Comment:
I think these should be mapped in ODW, so the curated layer matches the model. Has that been discussed?

Reply:
> I think these should be mapped in ODW, so the curated layer matches the model. Has that been discussed?

Yeah, agree this is something that should be in curated which should be coming soon. I've put this mapping here to get unblocked in the meantime, and it can be removed once the ODW mapping is in place